Topics/User_Experience/Mobile-First_Design.md

```diff
@@ -0,0 +1,54 @@
+* Prioritized content
+    * With a smaller screen, the UI for mobile devices will likely have to cut out any unnecessary details or rearrange content, allowing your designs to be cleaner and more concise
+    * For instance, having a large block of text that spans your entire phone screen might lead you to shorten the text or break it up for an improved UX; the lengthiness of that body of text might have been overlooked initially if using the desktop-first approach
+* Improved accessibility
+    * Mobile-first design emphasizes legible typography and text sizes, ensuring readability and navigability for your web app
+* Reduced redesigning
+    * Adopting this approach can reduce the need for extensive redesign efforts later in the process if some critical features on the desktop version, such as hover effects, cannot be easily translated into the mobile version
+* Faster page loading times
+    * With a reduced amount of code, images, and resources required for the web app, losing speed can also be reduced
+
+## Drawbacks
+
+* Learning curve
+    * Most web app developers are accustomed to the desktop-first approach, thus shifting to this new strategy that requires a different way of thinking and planning requires some effort
+* Restricted design creativity and flexibility
+    * Mobile designs are more limited than desktop versions, which may put more restrictions on the content and features you can include
+    * These constraints stem from the limited screen size and reliance on touch interactions, as designing the mobile version requires meticulous consideration to arrange the content in a touch-friendly and legible manner; this involves ensuring the buttons are sized appropriately for touch, accommodating all text within the narrow screen, and also minimizing the need for excessive scrolling
+
+## General Steps
+
+1. Content inventory
+    * Plan out all the content and elements you need for your web pages
+2. Visual hierarchy
+    * Create a priority list of all the visuals you will include on the page, taking into consideration the size of images, colours, and typography, as these properties can guide the user’s eye to understand the relative significance of the elements
+3. Design from smallest to largest screens
+    * Create designs for the mobile version first, then tablet, then finally desktop
+    * These designs can be made on Figma, Canva, Google Drawings/Slides, or whatever design tool that will help you plan out the exact format of your page
+4. Start coding!
+    * Use CSS media queries to adjust the size or style of elements depending on the screen size
+    * Test your UI on Google Chrome using the dev tool (right click on the screen, select Inspect, at the top select your desired dimensions to see how your web app looks on different kinds of devices and screen sizes)
```

Comment:
That's a good suggestion! I think if another page already shows this image, then maybe it would be redundant for me to also include it. However, maybe I can add a link to another resource that explains how to use the Inspect tool on Google Chrome, and tools on other browsers as well.

Reply:
I agree, a link in your resources instead would be helpful as well!